cmd/cmd.go

```diff
@@ -133,10 +150,169 @@ func initConfig() (config.Config, string, error) {
 }
 
 func initClient(logger logging.Logger, cfg config.Config) (pack.Client, error) {
-	client, err := pack.NewClient(pack.WithLogger(logger), pack.WithExperimental(cfg.Experimental), pack.WithRegistryMirrors(cfg.RegistryMirrors))
+	dc, err := initDockerClient()
+	if err != nil {
+		return pack.Client{}, err
+	}
+	client, err := pack.NewClient(pack.WithLogger(logger), pack.WithExperimental(cfg.Experimental), pack.WithRegistryMirrors(cfg.RegistryMirrors), pack.WithDockerClient(dc))
 	if err != nil {
 		return pack.Client{}, err
 	}
 
 	return *client, nil
 }
+
+func initDockerClient() (dockerClient.CommonAPIClient, error) {
+	dockerClientOpts := []dockerClient.Opt{
+		dockerClient.WithVersion("1.38"),
+	}
+
+	usesSSH := false
+	if dockerHost, ok := os.LookupEnv("DOCKER_HOST"); ok {
+		if _url, err := url.Parse(dockerHost); err == nil && _url.Scheme == "ssh" {
+			usesSSH = true
+			credentialsConfig := sshdialer.Config{
+				Identity:           os.Getenv("DOCKER_HOST_SSH_IDENTITY"),
+				PassPhrase:         os.Getenv("DOCKER_HOST_SSH_IDENTITY_PASSPHRASE"),
+				PasswordCallback:   newPasswordCbk(),
+				PassPhraseCallback: newPassPhraseCbk(),
+				HostKeyCallback:    newHostKeyCbk(),
+			}
+			dialer, err := sshdialer.NewDialer(_url, credentialsConfig)
+			if err != nil {
+				return nil, err
+			}
+
+			path := _url.Path
+			if path == "" {
+				path = "/var/run/docker.sock"
+			}
+
+			dialContext := func(ctx context.Context, network, addr string) (net.Conn, error) {
+				return dialer.DialContext(ctx, "unix", path)
+			}
+
+			httpClient := &http.Client{
+				// No tls
+				// No proxy
+				Transport: &http.Transport{
+					DialContext: dialContext,
+				},
+			}
+			dockerClientOpts = append(dockerClientOpts,
+				dockerClient.WithHTTPClient(httpClient),
+				dockerClient.WithHost("http://example.com/"),
+				dockerClient.WithDialContext(dialContext),
+			)
+		}
+	}
+	if !usesSSH {
+		dockerClientOpts = append(dockerClientOpts, dockerClient.FromEnv)
```

Comment:
How do we feel about `return nil if !usesSSH`? That way we could rely on the [fallback initializer](https://github.com/buildpacks/pack/blob/main/client.go#L164) and not have to duplicate these options? But then it would probably warrant a comment (_no strong feelings_)... 🤷🏾‍♂️

Reply:
Good point, I'll try that.